core/src/main/java/com/linecorp/armeria/client/retry/RetryingClient.java

```diff
@@ -65,4 +81,54 @@ protected Backoff newBackoff() {
     protected RetryStrategy<I, O> retryStrategy() {
         return retryStrategy;
     }
+
+    /**
+     * Returns the number of requests sent and increase its count by one.
+     */
+    protected int numRequestsSent(ClientRequestContext ctx) {
+        if (originalResponseTimeout == 0) {
+            originalResponseTimeout = ctx.responseTimeoutMillis();
```

Comment:
0 means 'timeout disabled'. Could you use a negative value? Also, could you check you are testing the case where the timeout is disabled?